src/mongo/MongoScrapbook.ts

```diff
@@ -204,4 +206,51 @@ class MongoScriptDocumentVisitor extends MongoVisitor<MongoCommand[]> {
 	protected defaultResult(_node: ParseTree): MongoCommand[] {
 		return this.commands;
 	}
+
+	private parseContext(ctx: mongoParser.ArgumentContext | mongoParser.PropertyValueContext): Object {
+		let parsedObject: Object = {};
+		if (!ctx || ctx.childCount === 0) { //Base case and malformed statements
+			return parsedObject;
+		}
+		// In a well formed expression, Argument and propertyValue tokens should have exactly one child, from their definitions in mongo.g4
+		// The only difference in types of children between PropertyValue and argument tokens is the functionCallContext that isn't handled at the moment.
+		let child: ParseTree = ctx.children[0];
+		if (child instanceof mongoParser.LiteralContext) {
+			let text = child.text;
+			let tokenType = child.start.type;
+			if (tokenType === mongoParser.mongoParser.StringLiteral) {
+				parsedObject = stripQuotes(text);
+			} else if (this._parserNonStringTokenTypes.indexOf(tokenType) > -1) {
+				parsedObject = JSON.parse(text);
+			}
+		}
+		else if (child instanceof mongoParser.ObjectLiteralContext) {
+			let propertyNameAndValue = randomUtils.findType(child.children, mongoParser.PropertyNameAndValueListContext);
+			if (!propertyNameAndValue) { // Argument is {}
+				return {};
+			}
+			else {
+				//tslint:disable:no-non-null-assertion
+				let propertyAssignments = randomUtils.filterType(propertyNameAndValue.children, mongoParser.PropertyAssignmentContext);
+				for (let propertyAssignment of propertyAssignments) {
+					const propertyName = <mongoParser.PropertyNameContext>propertyAssignment.children[0];
+					const propertyValue = <mongoParser.PropertyValueContext>propertyAssignment.children[2];
+					parsedObject[stripQuotes(propertyName.text)] = this.parseContext(propertyValue);
+				}
+			}
+		}
+		else if (child instanceof mongoParser.ArrayLiteralContext) {
+			let elementList = randomUtils.findType(child.children, mongoParser.ElementListContext);
+			let propertyValues = randomUtils.filterType(elementList.children, mongoParser.PropertyValueContext);
+			parsedObject = propertyValues.map(this.parseContext.bind(this));
+		} else {
+			console.assert("Unrecognized child type in parse tree.");
+		}
+		// tslint:disable-next-line:no-suspicious-comment
+		//TODO: else if(child instanceof mongoParser.FunctionCallContext)
+
+		return parsedObject;
+	}
+	private _parserNonStringTokenTypes = [mongoParser.mongoParser.NullLiteral, mongoParser.mongoParser.BooleanLiteral, mongoParser.mongoParser.NumericLiteral];
```

Comment:
This name doesn't adequately describe what this is (plenty of other non-string token types).  Personally I'd move this to the function where it's used and make it a const.

Reply:
Does `nonStringLiteral` sound appropriate?